src/gpu/intel/jit/codegen/kernel.hpp

```diff
@@ -188,62 +172,56 @@ struct setup_flags_t {
 
 setup_flags_t get_setup_flags(const stmt_t &s);
 
-template <typename ngen_generator_t>
-class ir_kernel_base_t : public ngen_generator_t {
+template <ngen::HW hw, typename BaseGeneratorT = generator_t<hw>>
```

Comment:
Why add the templating on hw? AFAICT, it just adds complexity here for almost no benefit. In particular, I am concerned that this will lead to accidental use of `hw` during codegen rather than the `getHardware()` interface which will be required for correct ngen_asm output.